broker-core/src/main/java/io/zeebe/broker/workflow/processor/activity/PropagateTerminationHandler.java

```diff
@@ -29,7 +29,9 @@
   public void handle(BpmnStepContext<ExecutableFlowNode> context) {
     final ElementInstance flowScopeInstance = context.getFlowScopeInstance();
 
-    if (flowScopeInstance.getChildren().isEmpty()) {
+    flowScopeInstance.consumeTokens(1);
+
+    if (flowScopeInstance.getActiveTokens() == 0) {
```

Comment:
Can it happen that we have negative tokens? Because we decremented to much?

Reply:
Should not be the case if all works fine :)

I agree though that the token counting (and understanding that it works correctly) is not obvious, especially since it is not in one place. I went with the current implementation and would like to implement parallel join first, and while doing that reiterate on it (for fork we need additional index state).